ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/DynamicPivotBlockManager.java

```diff
@@ -102,4 +102,11 @@ public void switchToNewPivotBlock(final BiConsumer<BlockHeader, Boolean> onSwitc
         },
         () -> onSwitchDone.accept(syncState.getPivotBlockHeader().orElseThrow(), false));
   }
+
+  public boolean isBlockchainBehind() {
+    return syncState
+        .getPivotBlockHeader()
+        .map(pivot -> syncActions.isBlockchainBehind(pivot.getNumber() - pivotBlockWindowValidity))
```

Comment:
why subtract the pivotBlockWindowValidity rather than just checking to see if the chain is behind the current pivot?

Reply:
I preferred to leave a safety window to avoid that we end up in a case where we will always be behind the pivot block. but I will do some tests without the 128 to see if it's ok. Thinking about it, I think it could pass because anyway the downloading of blocks is always faster than the creation of new blocks.

Reply:
testing on pre and post merge

post merge is ok just waiting for a mainnet sync to finish before merging